### PR TITLE
refactors various deconstructing procs into deconstruct() on /obj/

### PR DESCRIPTION
**Pull request by harryob**
## About The Pull Request

things should more uniformly use deconstruct() over destroy() (lowercase), destroy_airlock, destroy_window... etc.

deconstruct() should be called in place of qdel where the object needs to be physically broken down and removed, including ex_act(), deconstruction attack_hand() steps, etc, etc

## Why It's Good For The Game

there was confusion on how to get rid of an object. it's 2am as i do this and i've probably fucked up the logic in a bunch of places, but

## Changelog

:cl:
refactor: the code for deconstruction is now (somewhat) neater
/:cl:

### CODE REVIEW

**Comment by harryob** [comment]
remembered i need to do all this for ex_act, too

File: code/game/objects/effects/decals/misc.dm [hunk 1]
```diff
@@ -84,10 +84,14 @@
 	playsound(src, 'sound/effects/metal_crash.ogg', 50, 1)
 	M.animation_attack_on(src)
 	M.visible_message(SPAN_DANGER("[M] slices [src] apart!"), SPAN_DANGER("You slice [src] apart!"))
-	robogibs(src)
-	qdel(src)
+	deconstruct(TRUE)
 	return XENO_ATTACK_ACTION
 
+/obj/effect/decal/mecha_wreckage/deconstruct(disassembled = TRUE)
+	if(disassembled)
```
Comment (Watermelon914) [inline code/game/objects/effects/decals/misc.dm:91]: should be negated

File: code/game/objects/effects/decals/misc.dm [hunk 2]
```diff
@@ -84,10 +84,14 @@
...
+		robogibs(src)
+	. = ..()
```
Comment (Watermelon914) [inline code/game/objects/effects/decals/misc.dm:93]: always do `return ..()` at the end of a proc

File: code/game/objects/structures/crates_lockers/largecrate.dm [hunk 3]
```diff
@@ -18,18 +18,25 @@
...
-	qdel(src)
+	deconstruct(TRUE)
+
+/obj/structure/largecrate/deconstruct(disassembled = TRUE)
+	if(disassembled)
+		if(parts_type)
+			new parts_type(loc, 2)
+	else
+		new /obj/item/stack/sheet/wood(loc)
+	. = ..()
```
Comment (Watermelon914) [inline code/game/objects/structures/crates_lockers/largecrate.dm:32]: same as above here

File: code/game/objects/structures/displaycase.dm [hunk 4]
```diff
@@ -21,11 +21,14 @@
...
-				occupied = 0
-			qdel(src)
+			deconstruct(FALSE)
 
+/obj/structure/displaycase/deconstruct(disassemble = TRUE)
+	if(!disassemble)
+		new /obj/item/shard(src.loc)
+	if (occupied)
+		occupied = 0
+	. = ..()
```
Comment (Watermelon914) [inline code/game/objects/structures/displaycase.dm:31]: same thing here

File: code/game/objects/structures/displaycase.dm [hunk 5]
```diff
@@ -21,11 +21,14 @@
 				src.health -= 15
 				src.healthcheck()
 		if(EXPLOSION_THRESHOLD_MEDIUM to INFINITY)
-			new /obj/item/shard( src.loc )
-			if (occupied)
-				occupied = 0
-			qdel(src)
+			deconstruct(FALSE)
 
+/obj/structure/displaycase/deconstruct(disassemble = TRUE)
```
Comment (Watermelon914) [inline code/game/objects/structures/displaycase.dm:26]: should be disassembled and not disassemble

File: code/game/objects/structures/fence.dm [hunk 6]
```diff
@@ -178,6 +175,11 @@
 		healthcheck(1, 1, user, W)
 		..()
 
+/obj/structure/fence/deconstruct(disassembled = TRUE)
+	if(disassembled)
+		new /obj/item/stack/rods(loc, 10)
+	. = ..()
```
Comment (Watermelon914) [inline code/game/objects/structures/fence.dm:181]: same thing here

File: code/game/objects/structures/girders.dm [hunk 7]
```diff
@@ -215,11 +215,15 @@
...
-		new /obj/item/stack/sheet/metal(loc, 2)
-		qdel(src)
+
 		return TRUE
 	return FALSE
 
+/obj/structure/girder/deconstruct(disassembled = TRUE)
+	if(disassembled)
+		new /obj/item/stack/sheet/metal(loc, 2)
+	. = ..()
```
Comment (Watermelon914) [inline code/game/objects/structures/girders.dm:225]: same thing here

File: code/game/objects/structures/inflatable.dm [hunk 8]
```diff
@@ -112,18 +112,22 @@
...
+			deconstruct(FALSE)
+
+
+/obj/structure/inflatable/deconstruct(disassembled = TRUE)
+	if(!disassembled)
+		new /obj/structure/inflatable/popped(loc)
+	else
+		var/obj/item/inflatable/R = new /obj/item/inflatable(loc)
+		src.transfer_fingerprints_to(R)
+	. = ..()
```
Comment (Watermelon914) [inline code/game/objects/structures/inflatable.dm:129]: same thing here

File: code/game/objects/structures/lattice.dm [hunk 9]
```diff
@@ -58,12 +58,15 @@
...
-		qdel(src)
-
+			to_chat(user, SPAN_NOTICE("Slicing lattice joints..."))
+		deconstruct()
 	return
 
+/obj/structure/lattice/deconstruct(disassembled = TRUE)
+	if(disassembled)
+		new /obj/item/stack/rods(src.loc)
+	. = ..()
```
Comment (Watermelon914) [inline code/game/objects/structures/lattice.dm:68]: same thing here, do this for the rest of them in your PR since I don't want to list them all out

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

File: code/game/atoms.dm [hunk 10]
```diff
@@ -200,8 +200,13 @@ directive is properly returned.
 /atom/proc/relaymove()
 	return
 
-/atom/proc/ex_act()
-	return
+/atom/proc/contents_explosion(severity)
+	for(var/atom/A in contents)
+		A.ex_act(severity)
+		CHECK_TICK
```
Comment (fira) [inline code/game/atoms.dm:206]: :waaaaa: why CHECK_TICK ? does this even pass static checking ??

File: code/game/machinery/telecomms/machine_interactions.dm [hunk 11]
```diff
@@ -89,12 +89,13 @@
...
 
-					// Create a machine frame and delete the current machine
-					var/obj/structure/machinery/constructable_frame/F = new
-					F.forceMove(src.loc)
-					qdel(src)
-
+/obj/structure/machinery/telecomms/deconstruct(disassembled = TRUE)
+	if(disassembled) // Create a machine frame and delete the current machine
+		var/obj/structure/machinery/constructable_frame/F = new
+		F.forceMove(src.loc)
```
Comment (fira) [inline code/game/machinery/telecomms/machine_interactions.dm:97]: you can just pass the location as first arg in the new above

**Comment by fira** [comment]
Waiting on rereview by Walter, fine by me

**Review by fira — APPROVED** [review]

**Pull request merged**